### DISCUSSION TITLE
Search operators without fields

### DISCUSSION THREAD

**Discussion by akriese** [BeanieODM/beanie#570] [Question]
Hi, I want to be able to use query operators like Eq, GT etc. without fields. One simple use case is, that I want to use ElemMatch on a list of numbers. Like so:
```python
some_id: PydanticObjectId = ...
ElemMatch(UserRelation.users, Eq(some_id)) # UserRelation.users is a list of ids
# instead of
ElemMatch(UserRelation.users, {"$eq": some_id}) # this doesnt seem very beanie'ish to me :)

# or
threshold: int = 50
ElemMatch(User.bucket, GT(50))
# instead of
ElemMatch(User.bucket, {"$gt": 50}) # same here as above
```

In both cases, we can emit the field name, as it is about a simple element. The mongoDB queries would look like this:
```
{ "users": { "$elemMatch": { "$eq": some_id }}}
and
{ "bucket": { "$elemMatch": { "$gt": 50 }}}
```
where both times, the operator part doesnt include a field name.

So finally, my question is, how you would tackle this small problem. Would you just go with the inline query dict, or is there another way of doing this in beanie or even in mongoDB query?

Alternatively, I am thinking of making the field parameter of `BaseFindComparisonOperator` optional and if it is not given, then the query dict is constructed without it:
```python
class BaseFindComparisonOperator(BaseFindOperator):
    operator = ""

    def __init__(
        self,
        field = None,
        other = None,
    ):
        self.field = field
        self.other = other

    @property
    def query(self):
        inner = {self.operator: self.other}
        if self.field is None:
            return inner
        return {self.field: inner}
```

What do you think? Happy to hear your opinion :)

**Comment by roman-right** [comment]
Hi! Yup, eq operator and others can be different based on the context. I have to think about the pythonic syntax for it. Good catch! I'll convert it to the feature-request issue to not forget about it
